src/rollsum.c

```diff
@@ -26,26 +26,28 @@
 #define DO4(buf,i)  DO2(buf,i); DO2(buf,i+2);
 #define DO8(buf,i)  DO4(buf,i); DO4(buf,i+4);
 #define DO16(buf)   DO8(buf,0); DO8(buf,8);
-#define OF16(off)  {s1 += 16*off; s2 += 136*off;}
 
 void RollsumUpdate(Rollsum *sum,const unsigned char *buf,unsigned int len) {
-    /* ANSI C says no overflow for unsigned. 
-     zlib's adler 32 goes to extra effort to avoid overflow*/
+    /* ANSI C says no overflow for unsigned. zlib's adler32 goes to extra
+    effort to avoid overflow for its mod prime, which we don't have.*/
+    unsigned int n = len;
```

Comment:
It'd be better if you left the param `len` unchanged through the function and counted down in `n`.